### PR TITLE
Ditto: Fix manifest

### PR DESCRIPTION
**Pull request by niheaven**
Current manifest of Ditto is a broken one, it's not portable and the `persist` is wrong.

The fixes are:

- Change `url` from installer to portable zip
- `homepage` move to new one
- Add `autoupdate.hash`
- Fix `persist` by some slightly complicated hardcode (start ditto.exe, wait until it created ditto.db, kill ditto.exe)

Now it works fine, at least for me.

### CODE REVIEW

File: ditto.json [hunk 1]
```diff
@@ -1,38 +1,59 @@
...
+        "    while (!(Test-Path \"$dir\\Ditto.db\")) {",
+        "        Start-Sleep -Milliseconds 500",
+        "    }",
+        "    Start-Sleep 1",
+        "    Stop-Process -Name Ditto",
+        "}",
+        "$file = 'Ditto.Settings'",
+        "if (!(Test-Path \"$persist_dir\\$file\")) {",
+        "    Write-Host 'File' $file 'does not exists. Creating.' -f Yellow",
+        "    $CONT = @('[Ditto]\r\nDBPath3=Ditto.db')",
```
Comment (Ash258) [inline ditto.json]:
```text
[Ditto]\r\nDBPath...
```

Will be in final file. 

Make it `@('[Ditto]', 'DBPath3=Ditto.db')` appending line ending is done in L37

File: ditto.json [hunk 2]
```diff
@@ -1,38 +1,59 @@
...
             "32bit": {
-                "url": "https://downloads.sourceforge.net/project/ditto-cp/Ditto/$version/DittoSetup_$underscoreVersion.exe"
+                "url": "https://downloads.sourceforge.net/project/ditto-cp/Ditto/$version/DittoPortable_$underscoreVersion.zip"
             },
             "64bit": {
-                "url": "https://downloads.sourceforge.net/project/ditto-cp/Ditto/$version/DittoSetup_64bit_$underscoreVersion.exe"
+                "url": "https://downloads.sourceforge.net/project/ditto-cp/Ditto/$version/DittoPortable_64bit_$underscoreVersion.zip"
             }
+        },
+        "hash": {
```
Comment (Ash258) [inline ditto.json]: It's downloding from sourceforge, so hash extraction is done automatically using sourceforge mode.

File: ditto.json [hunk 3]
```diff
@@ -1,38 +1,59 @@
 {
     "version": "3.22.20.0",
-    "extract_dir": "Ditto",
+    "homepage": "https://ditto-cp.sourceforge.io/",
```
Comment (Ash258) [inline ditto.json]: No description

File: ditto.json (hunk 1 above)
Comment (niheaven) [reply ditto.json]: Fixed.

File: ditto.json (hunk 2 above)
Comment (niheaven) [reply ditto.json]: This mode is not mentioned in the wiki... Fixed.

File: ditto.json (hunk 3 above)
Comment (niheaven) [reply ditto.json]: Added description.

**Pull request merged**